tests/unit/commands/validate/test_cli.py

```diff
@@ -50,13 +51,36 @@ def test_template_fails_validation(self, read_sam_file_patch, click_patch, templ
     @patch('samcli.commands.validate.validate.SamTemplateValidator')
     @patch('samcli.commands.validate.validate.click')
     @patch('samcli.commands.validate.validate._read_sam_file')
-    def test_template_passes_validation(self, read_sam_file_patch, click_patch, template_valiadator):
+    def test_template_passes_validation(self, read_sam_file_patch, click_patch, template_validator):
         template_path = 'path_to_template'
         read_sam_file_patch.return_value = {"a": "b"}
 
         is_valid_mock = Mock()
         is_valid_mock.is_valid.return_value = True
-        template_valiadator.return_value = is_valid_mock
+        template_validator.return_value = is_valid_mock
 
         do_cli(ctx=None,
                template=template_path)
+
+    @patch('samcli.commands.validate.validate._get_boto_client')
+    @patch('samcli.commands.validate.validate.SamTemplateValidator')
+    @patch('samcli.commands.validate.validate.click')
+    @patch('samcli.commands.validate.validate._read_sam_file')
+    def test_profile_is_passed(self, read_sam_file_patch, click_patch, template_validator, get_boto_client_patch):
+        template_path = 'path_to_template'
+        profile = 'development'
+        read_sam_file_patch.return_value = {"a": "b"}
+
+        is_valid_mock = Mock()
+        is_valid_mock.is_valid.return_value = True
+        template_validator.return_value = is_valid_mock
+
+        # We were relying on an actual implementation in tests. Changing that seems
+        # highly out of scope for the current change. I'm limiting the scope to
+        # breaking dependency on the profile being defined in the runtime environment.
+        # If I understand the code correctly though, this is a major red flag for testability.
+        get_boto_client_patch.return_value = boto3.client('iam')
```

Comment:
I do not understand this comment. Can you explain this more to me, so I understand what you mean? I think it is referring to your comment in the description of the PR?

`boto3.client('iam')` should be mocked, otherwise the client is actually created (which it shouldn't in unit tests).